Initial commit to include "label" variable in Atom class

pgbarletta opened the pull request:
In the `Atom` class:
-  `name_` to `element_`
- add `label_`, and modified/added corresponding functions/operators.

In PDB.cpp
- read `label` and construct Atom object with both `element_` and `label_`

Then added corresponding tests in tests/atoms.cpp

------

TODO: 
- check it.
- Replace `name` by `element` in the rest of the project.

Luthaf commented:
Thank you for the pull request! If you need help with git, Github, Travis or any other tool used here, let me know.

--

I'll give your changes a look tomorrow. Another point to consider is to update the C API (`chemfiles.h`) to use **element** instead of **name** and to expose the new `label`.

pgbarletta commented:
>  If you need help with git, Github, Travis or any other tool used here, let me know.

Yes, you guessed right. This is my first time in open source contributing.

I'll add `label` support tomorrow to "chemfiles.h" tomorrow.

Luthaf commented:
> Yes, you guessed right. This is my first time in open source contributing.

Welcome! 

Travis and Appveyor are services checking that the code compiles and that the tests run on multiple platforms (Travis is for Linux and OS X; Appveyor for Windows). They integrte with Github and set a status for each commit you add (the little red cross on the right). You can have a look at the build details with the Detail links, and get the build log for each individual build.

For faster work, you should always build the code and run the tests locally, to fix any issue you can have before sending it to Github.

pgbarletta commented:
I'm sorry for making so many failed commits, but this `name_` to `element_` affects many files and, for some reason, my local "make" doesn't detect any errors; so this is taking some time.

### Notable changes:

- commented :

`Atom(std::string element);`

in "Atom.hpp" and its implementation in "Atom.cpp":
```
Atom::Atom(std::string element) : element_(std::move(element)), mass_(0), charge_(0) {
    if (is_element(element_)) {
        type_ = ELEMENT;
    } else {
        type_ = COARSE_GRAINED;
    }

    if (PERIODIC_INFORMATION.find(element_) != PERIODIC_INFORMATION.end()) {
        mass_ = PERIODIC_INFORMATION.at(element_).mass;
    }
}
```
to avoid the overlap with the new constructor that includes a default `label_`.

- Renamed `NamExpr` to `ElementExpr` in "expr.cpp" and "expr.hpp".

Luthaf commented:
> commented [...] to avoid the overlap with the new constructor that includes a default label_.

I think we can just remove this constructor, as the other one provides the exact same capacity.

> Renamed NamExpr to ElementExpr in "expr.cpp" and "expr.hpp".

I am not sure this is needed. We could keep the `NameExpr` and make it match on the atomic label. What do VMD do here? When selecting with `name H`, I get all atoms whose label is `H`?

--
>  for some reason, my local "make" doesn't detect any errors; so this is taking some time.

This is strange. What is you setup? (OS, compiler, cmake version)?

Luthaf commented on include/chemfiles/Atom.hpp:
```diff
@@ -37,29 +37,35 @@ class CHFL_EXPORT Atom {
+    //! Create an element from its `element` and its label
+    Atom(std::string element, std::string label = "");
+    //! Create an element from its `element`and its type
+    Atom(AtomType type, std::string element = "");
```
This constructor could get an additional `label` parameter.

Luthaf commented on include/chemfiles.h:
```diff
@@ -910,11 +910,11 @@ CHFL_EXPORT chfl_status chfl_residue_free(CHFL_RESIDUE* residue);
 /*!
-* @brief Create an atom from an atomic name
-* @param name The new atom name
+* @brief Create an atom from an atomic element
```
'atomic element' => 'element name'

Luthaf commented on include/chemfiles.h:
```diff
@@ -969,37 +969,37 @@ CHFL_EXPORT chfl_status chfl_atom_charge(const CHFL_ATOM* const atom, float* cha
 
 /*!
-* @brief Get the name of an atom
+* @brief Get the element of an atom
```
'element' => 'element name'

Luthaf commented on include/chemfiles.h:
```diff
@@ -969,37 +969,37 @@ CHFL_EXPORT chfl_status chfl_atom_charge(const CHFL_ATOM* const atom, float* cha
+* @brief Get the element of an atom
 * @param atom The atom
-* @param name A string buffer to be filled with the name
+* @param element A string buffer to be filled with the element
```
same

Luthaf commented on include/chemfiles.h:
```diff
@@ -969,37 +969,37 @@ CHFL_EXPORT chfl_status chfl_atom_charge(const CHFL_ATOM* const atom, float* cha
 
 /*!
-* @brief Set the name of an atom
+* @brief Set the element of an atom
```
same

Luthaf commented on include/chemfiles.h:
```diff
@@ -969,37 +969,37 @@ CHFL_EXPORT chfl_status chfl_atom_charge(const CHFL_ATOM* const atom, float* cha
+* @brief Set the element of an atom
 * @param atom The atom
-* @param name A null terminated string containing the new name
+* @param element A null terminated string containing the new element
```
same

Luthaf commented on include/chemfiles.h:
```diff
@@ -969,37 +969,37 @@ CHFL_EXPORT chfl_status chfl_atom_charge(const CHFL_ATOM* const atom, float* cha
 
 /*!
-* @brief Try to get the full name of an atom from the short name
+* @brief Try to get the full name of an atom from the element
```
same

Luthaf commented on include/chemfiles.h:
```diff
@@ -969,37 +969,37 @@ CHFL_EXPORT chfl_status chfl_atom_charge(const CHFL_ATOM* const atom, float* cha
+* @brief Try to get the full name of an atom from the element
 * @param atom The atom
-* @param name A string buffer to be filled with the name
+* @param element A string buffer to be filled with the element
```
same

Luthaf commented on include/chemfiles.h:
```diff
@@ -969,37 +969,37 @@ CHFL_EXPORT chfl_status chfl_atom_charge(const CHFL_ATOM* const atom, float* cha
 
 /*!
-* @brief Try to get the Van der Waals radius of an atom from the short name
+* @brief Try to get the Van der Waals radius of an atom from the element
```
same

Luthaf commented on include/chemfiles.h:
```diff
@@ -1009,7 +1009,7 @@ CHFL_EXPORT chfl_status chfl_atom_vdw_radius(const CHFL_ATOM* const atom,
 
 /*!
-* @brief Try to get the covalent radius of an atom from the short name
+* @brief Try to get the covalent radius of an atom from the element
```
same

Luthaf commented on include/chemfiles.h:
```diff
@@ -1019,7 +1019,7 @@ CHFL_EXPORT chfl_status chfl_atom_covalent_radius(const CHFL_ATOM* const atom,
 
 /*!
-* @brief Try to get the atomic number of an atom from the short name
+* @brief Try to get the atomic number of an atom from the element
```
same

Luthaf commented on include/chemfiles/Atom.hpp:
```diff
@@ -19,7 +19,7 @@ namespace chemfiles {
  *
  * An Atom is a particle in the current Frame. It can be used to store and
- * retrieve informations about a particle, such as mass, name, atomic number,
+ * retrieve informations about a particle, such as mass, element, atomic number,
```
same

Luthaf commented on include/chemfiles/Atom.hpp:
```diff
@@ -37,29 +37,35 @@ class CHFL_EXPORT Atom {
 
-    //! Set the atom name
-    void set_name(const std::string& name) { name_ = name; }
+    //! Set the atom element
```
same

Luthaf commented on tests/atoms.cpp:
```diff
@@ -10,22 +10,29 @@ TEST_CASE("Use the Atom type", "[Atoms]"){
         CHECK(a3.mass() == 0);
         CHECK(a3.charge() == 0);
+
+		CHECK(a5.type() == Atom::ELEMENT);
```
Nice to add more tests! Could you make sure the indentation is 4 spaces everywhere?

Luthaf commented:
OK, I messed with up the review tool ... `same` means `'element' => 'element name'` :smiley:

Luthaf commented on CMakeLists.txt:
```diff
@@ -17,15 +17,15 @@ string(REGEX REPLACE "^[0-9]+\\.([0-9]+).*" "\\1" CHEMFILES_VERSION_MINOR "${CHE
+option(CHFL_ENABLE_NETCDF "Enable AMBER NetCDF format." ON)
+option(CHFL_USE_WARNINGS "Compile the code with warnings (default in debug mode)" ON)
+option(CHFL_USE_SANITIZERS "Compile the code with sanitizers" ON)
+option(CHFL_DEBUG_GLIBCXX "Use the debug version of GNU libstdcxx" ON)
```
You do not need to change the value here, it is only the default. You can use `cmake -DCHFL_DEBUG_GLIBCXX=ON .` on an already configured build directpry to change a value, or `ccmake .` to start a graphical user interface. You also have `cmake-gui` to make these changes.

pgbarletta commented:
> I think we can just remove this constructor, as the other one provides the exact same capacity.

Removed constructor.

>I am not sure this is needed. We could keep the NameExpr and make it match on the atomic label. What do VMD do here? When selecting with name H, I get all atoms whose label is H?

Yes. In protein PDBs, "H" is the name for the hydrogen atom bonded to the alpha carbon (CA in PDB). If you type "name H" you will only get this particular kind of hydrogen, but if you type "element H" you will get all hydrogens.
I think I should include a "LabelExpr" to match what we call the `label_`.

>This is strange. What is you setup? (OS, compiler, cmake version)?
`Netrunner version of Ubuntu17,    g++ 5.2.1,   cmake 3.2.2` I'm not able to compile the C API. I'll keep using Travis CI until I fix it.

Changed "element" to "element name".  
Fixed weird indentation error in "tests/atoms.cpp". I use tab based indentations and it seems github and I have different thoughts on tab length.

Luthaf commented:
> Yes. In protein PDBs, "H" is the name for the hydrogen atom bonded to the alpha carbon (CA in PDB). If you type "name H" you will only get this particular kind of hydrogen, but if you type "element H" you will get all hydrogens.

So LabelExpr can still be called NameExpr and match on the label, and we should add `ElementExpr`? Or am I misunderstanding here?

In any case, I can do this, the selection code is a bit messy.

> I use tab based indentations and it seems github and I have different thoughts on tab length.

Yeah, Github uses 8 spaces for a tab. That is why I prefer using spaces for indentation: it looks always the same.

> Netrunner version of Ubuntu17, g++ 5.2.1, cmake 3.2.2 I'm not able to compile the C API.

What is your error message here? This should not happen.

Luthaf commented:
The code looks good to me. You need to fix the options (https://github.com/chemfiles/chemfiles/pull/41#discussion_r81203794) for Appveyor and Travis builds to complete.

pgbarletta commented:
Using `element_` and `label_` for atoms instead of `element_` and `name_`, was a good idea, since `Residue` class is the one that has the `name_` variable.  So I was thinking of this:

`ElementExpr` for Atom elements
`LabelExpr` for Atom labels
`NameExpr` for Residue names


> In any case, I can do this, the selection code is a bit messy.

Good, I wouldn't even know where to start.

----
This test keeps failing:
`CHECK( topology[0].element() == "ZN" )`
I can't find anything wrong in "PDB.cpp" nor "tests/pdb.cpp", also checked "MOF-5.pdb" and it looks properly formatted. Could you give me a hand with this?

---

TODO: 

- add `label_` support in C API and `Selections` class. Also add tests in tests/pdb.cpp and tests/topology.cpp.
- How about this?: read atom charge from PDB format. Columns 79-80. Add corresponding tests.

That's all I can think of right now.

Luthaf commented:
> So I was thinking of this:
> ElementExpr for Atom elements
> LabelExpr for Atom labels
> NameExpr for Residue names

I would rather use the same selectors that VMD, to prevent people from being confused when switching. So `name`, `element` and `resname/res`.

> This test keeps failing: `CHECK( topology[0].element() == "ZN" )`

I'll have a look.

> add label_ support in Selections class.

This can be in a separated PR.

> How about this?: read atom charge from PDB format. Columns 79-80. Add corresponding tests.

Why not. This can be a separated PR, taking care of still being able to read files without this information.

Luthaf commented on src/Atom.cpp:
```diff
@@ -27,55 +27,59 @@ static bool is_element(const std::string& name) {
-Atom::Atom(std::string name) : name_(std::move(name)), mass_(0), charge_(0) {
-    if (is_element(name_)) {
+Atom::Atom(std::string element, std::string label)
+	: element_(std::move(element)), label_(std::move(label)), mass_(0),
```
I would add a check here and in the other constructor to set the `label` to the same value as the `element` if the `label` is empty.

pgbarletta replied:
Instead of the conditional, I added a new `Atom` constructor method that only takes an "element" argument and assigns it to `element_` and `label_`, so the decision is made at compile time. Is this OK, or waaaay too much trouble for such a simple task?

Luthaf replied:
This is OK =)

Luthaf commented on src/formats/PDB.cpp:
```diff
@@ -284,7 +284,7 @@ void PDBFormat::write(const Frame& frame) {
         fmt::print(textfile_, "HETATM{:5d}{: >4s} {:3}X{:4d} "
                               "{:8.3f}{:8.3f}{:8.3f}{:6.2f}{:6.2f}{: >2s}\n",
-                   i, name, "", i, pos[0], pos[1], pos[2], 0.0, 0.0, name);
+                   i, element, "", i, pos[0], pos[1], pos[2], 0.0, 0.0, element);
```
The first element should be replaced by the atomic label now that we have it.

pgbarletta replied:
done

Luthaf commented:
> This test keeps failing: `CHECK( topology[0].element() == "ZN" )`

This was because the element name start at column 77 in the PDB format, but the PDB format uses 1-based indexing, so you need to use `auto atom = Atom(trim(line.substr(76, 2)), trim(line.substr(12, 4)));`.

After that, I get out of bound errors in `line.substr(76, 2)`, because we only check that the atom record is bigger than 54. If it is smaller than 78, I would use an empty string as the element name.

I also found two others issues in the code, seem my comments.

Luthaf commented on src/Atom.cpp:
```diff
@@ -36,6 +36,20 @@ static bool is_element(const std::string& element) {
 }
 
+Atom::Atom(std::string element)
+	: element_(std::move(element)), label_(std::move(element)), mass_(0),
```
You can not move the same object twice, this is undefined behaviour.

pgbarletta commented:
At last! (unless codecov/patch is critical)
I'll add the rest of the features later.

Luthaf commented:
> At last! (unless codecov/patch is critical)

It is not really, more an incentive to keep the code coverage high.

Luthaf commented:
>  I'll add the rest of the features later.

Do you want to add them in this PR, or should I go for the last review and merge?

pgbarletta commented:
I think I should add label support in the C API before merging. I'll do it
tonight

On Oct 3, 2016 8:30 AM, "Luthaf" <notifications@github.com> wrote:

> I'll add the rest of the features later.
>
> Do you want to add them in this PR, or should I go for the last review and
> merge?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/chemfiles/chemfiles/pull/41#issuecomment-251084292>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFbgK-lahctYL_7Fj7Vp_fzEjp0wk2CRks5qwOdbgaJpZM4KJFaM>
> .
>

pgbarletta commented:
These are the failed tests when I run `cmake`, that (I think) prevent me from compiling the C API:
```
-- Performing Test CC_SUPPORTS-Wsign-promo - Failed
-- Performing Test CC_SUPPORTS-Wsuggest-override - Failed
-- Performing Test CXX_SUPPORTS-Wdocumentation - Failed
-- Performing Test CC_SUPPORTS-Wdocumentation - Failed
-- Performing Test CXX_SUPPORTS-Wextra-semi - Failed
-- Performing Test CC_SUPPORTS-Wextra-semi - Failed
-- Performing Test CC_SUPPORTS-Wnon-virtual-dtor - Failed
-- Performing Test CC_SUPPORTS-Wold-style-cast - Failed
-- Performing Test CC_SUPPORTS-Woverloaded-virtual - Failed
```

Some of them refer to documentation, but the rest of them refer to the C compiler. Maybe I should point `cmake` to use `clang` instead of `gcc`?

Luthaf commented:
> Some of them refer to documentation, but the rest of them refer to the C compiler. Maybe I should point cmake to use clang instead of gcc?

No, these are tests to check which warning flags are supported by your version of the compiler. It should not prevent the code from compiling at all.

Can you post a full log of doing
```
cmake ..
make
```
from a clean `build` directory?

pgbarletta commented:
I started from scratch and C compilation seems to be working. I also commented out a few lines of "capi/atom.cpp" and got the corresponding compilation error, unlike before.


-------------------

```
git clone https://github.com/chemfiles/chemfiles
cd chemfiles
mkdir build
cd build
cmake .. -DCMAKE_INSTALL_PREFIX=/home/german/tempinc &> logs
make &> make_logs
make install &> install_logs
```

and got this:


```
-- The C compiler identification is GNU 5.2.1
-- The CXX compiler identification is GNU 5.2.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Success
-- Performing Test COMPILER_SUPPORTS_CXX0X
-- Performing Test COMPILER_SUPPORTS_CXX0X - Success
-- Performing Test COMPILER_SUPPORTS_C99
-- Performing Test COMPILER_SUPPORTS_C99 - Success
-- Performing Test CXX_SUPPORTS-Wall
-- Performing Test CXX_SUPPORTS-Wall - Success
-- Performing Test CC_SUPPORTS-Wall
-- Performing Test CC_SUPPORTS-Wall - Success
-- Performing Test CXX_SUPPORTS-Wextra
-- Performing Test CXX_SUPPORTS-Wextra - Success
-- Performing Test CC_SUPPORTS-Wextra
-- Performing Test CC_SUPPORTS-Wextra - Success
-- Performing Test CXX_SUPPORTS-Wuninitialized
-- Performing Test CXX_SUPPORTS-Wuninitialized - Success
-- Performing Test CC_SUPPORTS-Wuninitialized
-- Performing Test CC_SUPPORTS-Wuninitialized - Success
-- Performing Test CXX_SUPPORTS-Wconversion
-- Performing Test CXX_SUPPORTS-Wconversion - Success
-- Performing Test CC_SUPPORTS-Wconversion
-- Performing Test CC_SUPPORTS-Wconversion - Success
-- Performing Test CXX_SUPPORTS-Wsign-conversion
-- Performing Test CXX_SUPPORTS-Wsign-conversion - Success
-- Performing Test CC_SUPPORTS-Wsign-conversion
-- Performing Test CC_SUPPORTS-Wsign-conversion - Success
-- Performing Test CXX_SUPPORTS-Wsign-promo
-- Performing Test CXX_SUPPORTS-Wsign-promo - Success
-- Performing Test CC_SUPPORTS-Wsign-promo
-- Performing Test CC_SUPPORTS-Wsign-promo - Failed
-- Performing Test CXX_SUPPORTS-Wsuggest-override
-- Performing Test CXX_SUPPORTS-Wsuggest-override - Success
-- Performing Test CC_SUPPORTS-Wsuggest-override
-- Performing Test CC_SUPPORTS-Wsuggest-override - Failed
-- Performing Test CXX_SUPPORTS-Wsuggest-final-types
-- Performing Test CXX_SUPPORTS-Wsuggest-final-types - Success
-- Performing Test CC_SUPPORTS-Wsuggest-final-types
-- Performing Test CC_SUPPORTS-Wsuggest-final-types - Success
-- Performing Test CXX_SUPPORTS-Wpedantic
-- Performing Test CXX_SUPPORTS-Wpedantic - Success
-- Performing Test CC_SUPPORTS-Wpedantic
-- Performing Test CC_SUPPORTS-Wpedantic - Success
-- Performing Test CXX_SUPPORTS-pedantic
-- Performing Test CXX_SUPPORTS-pedantic - Success
-- Performing Test CC_SUPPORTS-pedantic
-- Performing Test CC_SUPPORTS-pedantic - Success
-- Performing Test CXX_SUPPORTS-Wdocumentation
-- Performing Test CXX_SUPPORTS-Wdocumentation - Failed
-- Performing Test CC_SUPPORTS-Wdocumentation
-- Performing Test CC_SUPPORTS-Wdocumentation - Failed
-- Performing Test CXX_SUPPORTS-Wdeprecated
-- Performing Test CXX_SUPPORTS-Wdeprecated - Success
-- Performing Test CC_SUPPORTS-Wdeprecated
-- Performing Test CC_SUPPORTS-Wdeprecated - Success
-- Performing Test CXX_SUPPORTS-Wextra-semi
-- Performing Test CXX_SUPPORTS-Wextra-semi - Failed
-- Performing Test CC_SUPPORTS-Wextra-semi
-- Performing Test CC_SUPPORTS-Wextra-semi - Failed
-- Performing Test CXX_SUPPORTS-Wnon-virtual-dtor
-- Performing Test CXX_SUPPORTS-Wnon-virtual-dtor - Success
-- Performing Test CC_SUPPORTS-Wnon-virtual-dtor
-- Performing Test CC_SUPPORTS-Wnon-virtual-dtor - Failed
-- Performing Test CXX_SUPPORTS-Wold-style-cast
-- Performing Test CXX_SUPPORTS-Wold-style-cast - Success
-- Performing Test CC_SUPPORTS-Wold-style-cast
-- Performing Test CC_SUPPORTS-Wold-style-cast - Failed
-- Performing Test CXX_SUPPORTS-Wcast-align
-- Performing Test CXX_SUPPORTS-Wcast-align - Success
-- Performing Test CC_SUPPORTS-Wcast-align
-- Performing Test CC_SUPPORTS-Wcast-align - Success
-- Performing Test CXX_SUPPORTS-Wunused
-- Performing Test CXX_SUPPORTS-Wunused - Success
-- Performing Test CC_SUPPORTS-Wunused
-- Performing Test CC_SUPPORTS-Wunused - Success
-- Performing Test CXX_SUPPORTS-Woverloaded-virtual
-- Performing Test CXX_SUPPORTS-Woverloaded-virtual - Success
-- Performing Test CC_SUPPORTS-Woverloaded-virtual
-- Performing Test CC_SUPPORTS-Woverloaded-virtual - Failed
-- Performing Test CXX_SUPPORTS-Wno-unknown-pragmas
-- Performing Test CXX_SUPPORTS-Wno-unknown-pragmas - Success
-- Performing Test CC_SUPPORTS-Wno-unknown-pragmas
-- Performing Test CC_SUPPORTS-Wno-unknown-pragmas - Success
-- Setting build type to 'release' as none was specified.
-- Unpacking molfiles sources
-- Found NetCDF: /usr/lib/x86_64-linux-gnu/libnetcdf.so  
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Configuring done
-- Generating done
-- Build files have been written to: /home/german/tempchem/chemfiles/build
```
CMake output (logs)
```
Scanning dependencies of target fmt
[  2%] Building CXX object external/CMakeFiles/fmt.dir/fmt/fmt/format.cc.o
[  5%] Building CXX object external/CMakeFiles/fmt.dir/fmt/fmt/posix.cc.o
[  8%] Building CXX object external/CMakeFiles/fmt.dir/fmt/fmt/ostream.cc.o
[  8%] Built target fmt
Scanning dependencies of target molfiles
[ 11%] Building C object external/CMakeFiles/molfiles.dir/molfiles/src/dcdplugin.c.o
[ 14%] Building CXX object external/CMakeFiles/molfiles.dir/molfiles/src/gromacsplugin.cxx.o
[ 17%] Building C object external/CMakeFiles/molfiles.dir/molfiles/src/pdbplugin.c.o
[ 20%] Building C object external/CMakeFiles/molfiles.dir/molfiles/src/lammpsplugin.c.o
[ 20%] Built target molfiles
Scanning dependencies of target chemfiles
[ 22%] Building CXX object CMakeFiles/chemfiles.dir/src/Connectivity.cpp.o
[ 25%] Building CXX object CMakeFiles/chemfiles.dir/src/files/BasicFile.cpp.o
[ 28%] Building CXX object CMakeFiles/chemfiles.dir/src/files/NcFile.cpp.o
[ 31%] Building CXX object CMakeFiles/chemfiles.dir/src/Topology.cpp.o
[ 34%] Building CXX object CMakeFiles/chemfiles.dir/src/UnitCell.cpp.o
[ 37%] Building CXX object CMakeFiles/chemfiles.dir/src/capi/selection.cpp.o
[ 40%] Building CXX object CMakeFiles/chemfiles.dir/src/capi/residue.cpp.o
[ 42%] Building CXX object CMakeFiles/chemfiles.dir/src/capi/frame.cpp.o
[ 45%] Building CXX object CMakeFiles/chemfiles.dir/src/capi/logging.cpp.o
[ 48%] Building CXX object CMakeFiles/chemfiles.dir/src/capi/topology.cpp.o
[ 51%] Building CXX object CMakeFiles/chemfiles.dir/src/capi/atom.cpp.o
[ 54%] Building CXX object CMakeFiles/chemfiles.dir/src/capi/cell.cpp.o
[ 57%] Building CXX object CMakeFiles/chemfiles.dir/src/capi/trajectory.cpp.o
[ 60%] Building CXX object CMakeFiles/chemfiles.dir/src/selections/expr.cpp.o
[ 62%] Building CXX object CMakeFiles/chemfiles.dir/src/selections/lexer.cpp.o
[ 65%] Building CXX object CMakeFiles/chemfiles.dir/src/selections/parser.cpp.o
[ 68%] Building CXX object CMakeFiles/chemfiles.dir/src/Trajectory.cpp.o
[ 71%] Building CXX object CMakeFiles/chemfiles.dir/src/Atom.cpp.o
[ 74%] Building CXX object CMakeFiles/chemfiles.dir/src/Logger.cpp.o
[ 77%] Building CXX object CMakeFiles/chemfiles.dir/src/Residue.cpp.o
[ 80%] Building CXX object CMakeFiles/chemfiles.dir/src/formats/PDB.cpp.o
[ 82%] Building CXX object CMakeFiles/chemfiles.dir/src/formats/Molfile.cpp.o
[ 85%] Building CXX object CMakeFiles/chemfiles.dir/src/formats/NcFormat.cpp.o
[ 88%] Building CXX object CMakeFiles/chemfiles.dir/src/formats/XYZ.cpp.o
[ 91%] Building CXX object CMakeFiles/chemfiles.dir/src/TrajectoryFactory.cpp.o
[ 94%] Building CXX object CMakeFiles/chemfiles.dir/src/Frame.cpp.o
[ 97%] Building CXX object CMakeFiles/chemfiles.dir/src/Format.cpp.o
[100%] Building CXX object CMakeFiles/chemfiles.dir/src/Selections.cpp.o
Linking CXX static library libchemfiles.a
[100%] Built target chemfiles
```
Make output (make_logs)

```
[  8%] Built target fmt
[ 20%] Built target molfiles
[100%] Built target chemfiles
Install the project...
-- Install configuration: "release"
-- Installing: /home/german/tempinc/include/chemfiles/Selections.hpp
-- Installing: /home/german/tempinc/include/chemfiles/UnitCell.hpp
-- Installing: /home/german/tempinc/include/chemfiles/NcFile.hpp
-- Installing: /home/german/tempinc/include/chemfiles/BasicFile.hpp
-- Installing: /home/german/tempinc/include/chemfiles/Trajectory.hpp
-- Installing: /home/german/tempinc/include/chemfiles/Connectivity.hpp
-- Installing: /home/german/tempinc/include/chemfiles/Error.hpp
-- Installing: /home/german/tempinc/include/chemfiles/TrajectoryFactory.hpp
-- Installing: /home/german/tempinc/include/chemfiles/Residue.hpp
-- Installing: /home/german/tempinc/include/chemfiles/Topology.hpp
-- Installing: /home/german/tempinc/include/chemfiles/config.in.hpp
-- Installing: /home/german/tempinc/include/chemfiles/periodic.hpp
-- Installing: /home/german/tempinc/include/chemfiles/Format.hpp
-- Installing: /home/german/tempinc/include/chemfiles/sorted_set.hpp
-- Installing: /home/german/tempinc/include/chemfiles/lexer.hpp
-- Installing: /home/german/tempinc/include/chemfiles/parser.hpp
-- Installing: /home/german/tempinc/include/chemfiles/expr.hpp
-- Installing: /home/german/tempinc/include/chemfiles/optional.hpp
-- Installing: /home/german/tempinc/include/chemfiles/Atom.hpp
-- Installing: /home/german/tempinc/include/chemfiles/File.hpp
-- Installing: /home/german/tempinc/include/chemfiles/Logger.hpp
-- Installing: /home/german/tempinc/include/chemfiles/types.hpp
-- Installing: /home/german/tempinc/include/chemfiles/span.hpp
-- Installing: /home/german/tempinc/include/chemfiles/NcFormat.hpp
-- Installing: /home/german/tempinc/include/chemfiles/PDB.hpp
-- Installing: /home/german/tempinc/include/chemfiles/XYZ.hpp
-- Installing: /home/german/tempinc/include/chemfiles/Molfile.hpp
-- Installing: /home/german/tempinc/include/chemfiles/utils.hpp
-- Installing: /home/german/tempinc/include/chemfiles/Frame.hpp
-- Installing: /home/german/tempinc/include/chemfiles/capi.hpp
-- Installing: /home/german/tempinc/include/chemfiles/exports.hpp
-- Installing: /home/german/tempinc/include/chemfiles/config.hpp
-- Installing: /home/german/tempinc/include/chemfiles.hpp
-- Installing: /home/german/tempinc/include/chemfiles.h
-- Installing: /home/german/tempinc/lib/libchemfiles.a
```
Make install output (install_logs).

Luthaf commented:
Can you build the tests by setting `CHFL_BUILD_TESTS` to ON? (either using `ccmake` graphical interface, or `cmake -DCHFL_BUILD_TESTS=ON .`) and then running them with `ctest` ?

pgbarletta commented:
Worked just fine:
```
Test project /home/german/chemfiles/build
      Start  1: unit-cell
 1/29 Test  #1: unit-cell ........................   Passed    0.00 sec
      Start  2: residue
 2/29 Test  #2: residue ..........................   Passed    0.00 sec
      Start  3: frame
 3/29 Test  #3: frame ............................   Passed    0.00 sec
      Start  4: logging
 4/29 Test  #4: logging ..........................   Passed    0.00 sec
      Start  5: types
 5/29 Test  #5: types ............................   Passed    0.00 sec
      Start  6: atoms
 6/29 Test  #6: atoms ............................   Passed    0.00 sec
      Start  7: factory
 7/29 Test  #7: factory ..........................   Passed    0.00 sec
      Start  8: selections
 8/29 Test  #8: selections .......................   Passed    0.01 sec
      Start  9: topology
 9/29 Test  #9: topology .........................   Passed    0.00 sec
      Start 10: parser
10/29 Test #10: parser ...........................   Passed    0.01 sec
      Start 11: trajectory
11/29 Test #11: trajectory .......................   Passed    0.01 sec
      Start 12: ncfile
12/29 Test #12: ncfile ...........................   Passed    0.00 sec
      Start 13: basic-file
13/29 Test #13: basic-file .......................   Passed    0.01 sec
      Start 14: xtc-molfile
14/29 Test #14: xtc-molfile ......................   Passed    0.31 sec
      Start 15: dcd-molfile
15/29 Test #15: dcd-molfile ......................   Passed    0.00 sec
      Start 16: lammps-molfile
16/29 Test #16: lammps-molfile ...................   Passed    0.02 sec
      Start 17: xyz
17/29 Test #17: xyz ..............................   Passed    0.12 sec
      Start 18: pdb
18/29 Test #18: pdb ..............................   Passed    0.01 sec
      Start 19: trr-molfile
19/29 Test #19: trr-molfile ......................   Passed    0.04 sec
      Start 20: amber-netcdf
20/29 Test #20: amber-netcdf .....................   Passed    0.00 sec
      Start 21: gro-molfile
21/29 Test #21: gro-molfile ......................   Passed    0.01 sec
      Start 22: c-frame
22/29 Test #22: c-frame ..........................   Passed    0.00 sec
      Start 23: c-trajectory
23/29 Test #23: c-trajectory .....................   Passed    0.01 sec
      Start 24: c-logging
24/29 Test #24: c-logging ........................   Passed    0.00 sec
      Start 25: c-cell
25/29 Test #25: c-cell ...........................   Passed    0.00 sec
      Start 26: c-residue
26/29 Test #26: c-residue ........................   Passed    0.00 sec
      Start 27: c-atom
27/29 Test #27: c-atom ...........................   Passed    0.00 sec
      Start 28: c-selections
28/29 Test #28: c-selections .....................   Passed    0.00 sec
      Start 29: c-topology
29/29 Test #29: c-topology .......................   Passed    0.00 sec

100% tests passed, 0 tests failed out of 29

Total Test time (real) =   0.62 sec
```
Just a detail. CMake options lack the`CHFL` in the documentation:
http://chemfiles.readthedocs.io/en/latest/installation.html

Luthaf commented:
> Just a detail. CMake options lack theCHFL in the documentation: http://chemfiles.readthedocs.io/en/latest/installation.html

The documentation was moved to http://chemfiles.github.io/chemfiles/, so the one on readthedocs was outdated. I removed the project from readthedocs, do you know any link still pointing there?

pgbarletta commented:
Nope. But I'll let you know if I find one.
